### PR TITLE
Clean up logging and log failed API requests

### PR DESCRIPTION
**Pull request by divolgin**
#### What type of PR is this?

Cleaning up some noisy logging, terminal logging, and adding logger to mux router.

kind/chore

#### What this PR does / why we need it:

The goal is to improve information contained in support bundles.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

See comments

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Does this PR require documentation?

NONE

### CODE REVIEW

File: pkg/handlers/middleware.go [hunk 1]
```diff
@@ -21,23 +35,46 @@ func CorsMiddleware(next http.Handler) http.Handler {
 	})
 }
 
+func DebugLoggingMiddleware(next http.Handler) http.Handler {
```
Comment (divolgin) [inline pkg/handlers/middleware.go:38]: Made this separate for endpoints like healthz, so errors are never logged unless `DEBUG` is enabled.

File: kustomize/overlays/dev/deployment.yaml [hunk 2]
```diff
@@ -72,6 +72,8 @@ spec:
               value: "3"
             - name: POD_OWNER_KIND
               value: "deployment"
+            - name: DEBUG
+              value: "false"
```
Comment (sgalsaleh) [inline kustomize/overlays/dev/deployment.yaml:76]: but this is for the dev env, should it be set to true? i guess we can toggle it, but devs won't know this thing exists

Comment (divolgin) [reply kustomize/overlays/dev/deployment.yaml:76]: I just left it as a place holder for dev when I was testing this.  Made it 2% easier to toggle :)

**Comment by divolgin** [comment]
Some examples

#### before
```
  •  ✓  ing upstream
  •  ✓  ting base
  • Creating midstreams    • Transferring image nginx:latestnot cleaning up job "post-install-job", active=0, succeeeded=1, failed=0
not cleaning up job "pre-install-job", active=0, succeeeded=1, failed=0
{"level":"debug","ts":"2021-09-23T15:17:24Z","msg":"preflight analyze phase[]"}
{"level":"debug","ts":"2021-09-23T15:17:24Z","msg":"{cluster-resources completed 3 3}"}
{"level":"debug","ts":"2021-09-23T15:17:24Z","msg":"preflight marshalling[]"}
{"level":"debug","ts":"2021-09-23T15:17:24Z","msg":"preflight checks completed[]"}
    •  ✓  
    • Transferring image alpine:3.4not cleaning up job "post-install-job", active=0, succeeeded=1, failed=0
not cleaning up job "pre-install-job", active=0, succeeeded=1, failed=0
    •  ✓  
```
#### after
```
  • Pulling upstream
  •  ✓
  • Creating base
  •  ✓
  • Creating midstreams
I0923 15:43:33.976960       1 request.go:665] Waited for 1.017585897s due to client-side throttling, not priority and fairness, request: GET:https://10.43.0.1:443/apis/scheduling.k8s.io/v1?timeout=32s
    • Transferring image nginx:latest
    •  ✓
    • Transferring image alpine:3.4
    •  ✓
    • Transferring image curlimages/curl:latest
    •  ✓
    • Transferring image alpine
    •  ✓
    • Transferring image alpine
    •  ✓
    • Transferring image registry.staging.replicated.com/dmitriy-kots/netutils:latest
    •  ✓
    • Transferring image registry.staging.replicated.com/dmitriy-kots/netutils:latest
    •  ✓
  • Creating downstream "this-cluster"
  •  ✓
  • Creating downstream "this-cluster"
  •  ✓
  •  ✓
{"level":"debug","ts":"2021-09-23T15:45:04Z","msg":"checking for secrets replacers[]"}
```
#### before
```
not cleaning up job "post-install-job", active=0, succeeeded=1, failed=0
not cleaning up job "pre-install-job", active=0, succeeeded=1, failed=0
{"level":"error","ts":"2021-09-23T15:20:25Z","msg":"invalid aws secret access key"}


```
#### after
```
{"level":"error","ts":"2021-09-23T16:04:51Z","msg":"invalid aws secret access key"}
{"level":"info","ts":"2021-09-23T16:04:51Z","msg":"method=PUT status=400 duration=334.128351ms request=/api/v1/snapshots/settings"}
```

**Review by sgalsaleh — APPROVED** [review]

**Pull request merged**